src/Core/PersistedQueries.Abstractions/IWriteStoredQueries.cs

```diff
@@ -0,0 +1,19 @@
+using HotChocolate.Execution;
+using System.Threading.Tasks;
+
+namespace HotChocolate.PersistedQueries
+{
+    /// <summary>
+    /// A tool for storing queries to some persistence medium.
+    /// </summary>
+    public interface IStoreQueries
+    {
+        /// <summary>
+        /// Stores a given query using the given identifier.
+        /// </summary>
+        /// <param name="queryIdentifier">The query identifier.</param>
+        /// <param name="query">The query to store.</param>
+        /// <returns>An asynchronous operation.</returns>
+        Task StoreQueryAsync(string queryIdentifier, IQuery query);
```

Comment:
Same goes for this regarding symmetry of names.

Reply:
I originally went with `IWriteStoredQueries` and a method of `WriteStoredQueryAsync()` but something felt funny about using two words back-to-back that mean "persist". That said, I'm in no way opposed to going back to `IWriteStoredQueries`. I think there is still some symmetry between the interfaces in that one states "I _store_ queries" and then the other is "I read _stored_ queries" but it is a much looser symmetry than `IRead` and `IWrite`.